### PR TITLE
[@types/pino] Add optional generic to log function

### PR DESCRIPTION
**Pull request by runk**
Just means of providing a convenience in terms of enforcing specific type consumer is logging.

### CODE REVIEW

File: types/pino/index.d.ts [hunk 1]
```diff
@@ -733,6 +733,7 @@ declare namespace P {
          * Log at `'fatal'` level the given msg. If the first argument is an object, all its properties will be included in the JSON line.
          * If more args follows `msg`, these will be used to format `msg` using `util.format`.
          *
+         * @typeparam T: the interface of the object being serialized. Default is object.
```
Comment (lummish) [inline types/pino/index.d.ts:736]: `@typeparam` does not appear to be a valid tag according to the [Typedoc documentation](https://typedoc.org/guides/doccomments/#typeparam-param-name), assuming that's the system that parses these comments in the first place. Replace all instances with either `@typeParam` or `@template` please.

**Review by lummish — CHANGES_REQUESTED** [review]
Incorrect tag used to indicate generic type parameter.

Aside from this, I'm not sure that I see any reason to introduce this change. When and, more importantly, why would one explicitly specify the generic type when logging a message?

File: types/pino/index.d.ts (hunk 1 above)
Comment (runk) [reply types/pino/index.d.ts]: Fixed

**Comment by runk** [comment]
@lummish thanks for the review! 🙇  Can you please check it again?

**Comment by lummish** [comment]
@runk I'm not sure if you saw my review summary, but I'm not sure that I see any reason to introduce this change. When and, more importantly, why would one explicitly specify the generic type when logging a message?

**Comment by runk** [comment]
@lummish problem we're trying to solve is consistency in logged objects. E.g. making sure all logged objects have common field, let's say "correlationId". This makes troubleshooting and building reports a lot easier.

Obvious way to do that is to wrap all logging methods (`.info`, `.error` etc) with a fn on consumer side. Wondering if this is something that be done without ugly thin wrappers.

**Comment by ConradLang** [comment]
@lummish Further to what @runk mentioned, the consistency across multiple teams across the company means searching and correlating logs becomes a lot easier when trying to understand or diagnose requests and processes across all the services.

There's a lot less renaming / coalescing / manipulation in your log search tooling etc. meaning less effort looking for data and more ease analysing data.

**Comment by lummish** [comment]
@runk, @ConradLang makes sense. Will re-review.

**Review by lummish — APPROVED** [review]
LGTM

**Comment by runk** [comment]
Ready to merge

**Pull request merged**